src/main/kotlin/net/maizegenetics/phgv2/pathing/AlignmentUtils.kt

```diff
@@ -0,0 +1,292 @@
+
+private val myLogger = LogManager.getLogger("net.maizegenetics.phgv2.utils.AlignmentUtils")
+
+
+fun alignReadsToHaplotypes(hvcfDir: String, kmerIndexFile:String, keyFileRecords:List<KeyFileData>, paired:Boolean, outputDir:String) {
+    //loop through all files in hvcfDir and create a list of hvcf files
+    val hvcfFiles = File(hvcfDir).walkTopDown().filter { it.isFile }.filter { it.extension == "h.vcf" }.map { "${it.path}/${it.name}" }.toList()
+
+    //create a HaplotypeGraph from the list of hvcf files
+    val graph = HaplotypeGraph(hvcfFiles)
+
+    val kmerIndexMap = loadKmerMaps(kmerIndexFile, graph)
+
+
+}
+
+/**
+ * Loads a kmer hash map file to memory for use in mapping reads.
+ */
+fun loadKmerMaps(filename: String, graph: HaplotypeGraph): KmerMapData {
+    //Load the contents of the file into
+    //rangeHapidMap: a map of refRange to an Array<String> of the haplotype ids in the ReferenceRange
+    // and the BitSet of all hapid sets
+    //kmerHashmap: a map of kmer hash to reference range and offset into its BitSet, encoded as a long
+    //These data structures all the reference range and haplotype set to be looked up for a kmer has
+
+
+    val rangeToBitSetMap = mutableMapOf<ReferenceRange, BitSet>()
+    val kmerHashMap = Long2LongOpenHashMap()
+    var lineCount = 0
+    var totalLineCount = 0
+    var refrange = ReferenceRange("NULL", 0, 0)
+
+    val refRangeToIdMap = getReferenceRangeToIndexMap(graph)
+    getBufferedReader(filename).useLines {
+        it.forEach { inputStr ->
+            totalLineCount++
+
+            lineCount = when(inputStr.first()) {
+                '>' -> 1
+                else -> lineCount + 1
+            }
+
+            when (lineCount) {
+                1 -> {
+                    //line 1 is the range
+                    refrange = ReferenceRange.parse(inputStr.substring(1))
+                }
+                2-> {
+                    try {
+                        val parsedLine = inputStr.split(",")
+                        val myBitset = BitSet.valueOf(parsedLine.map { it.toLong() }.toLongArray())
+                        rangeToBitSetMap[refrange] = myBitset
+                    } catch(e: Exception) {
+                        println("error at line $totalLineCount for input = $inputStr")
+                        throw java.lang.IllegalArgumentException(e)
+                    }
+                }
+                3 -> {
+                    val refrangeLong = (refRangeToIdMap[refrange]!!.toLong()) shl 32
+                    val parsedLine = inputStr.split(",")
+                    //values are pairs of hash, offset (long,int)
+                    //add the pairs of entries to kmerHashMap
+                    for (datapair in parsedLine) {
+                        val hashOffset = datapair.split("@")
+
+                        if (hashOffset.size < 2) {
+                            println("improperly formatted datapair at line $totalLineCount")
```

Comment:
Do you need to "continue" here?  Or throw an error?  if you don't and hashOffset.size < 2,  this could cause an exception on line 94 below when it tries to access the second value in hashOffset.

Reply:
This came directly from @pjbradbury s code in PHG v1.  I will defer to him to answer this question.

Reply:
It looks like that was added for debugging. Throwing an exception at line 94 is appropriate. This line just shows where the exception occurred to make it easier to fix the file. It might make more sense to throw an exception here but the end result is the same.